File: Programlar/Task 1/123.cpp
#include<iostream>
using namespace std;
bool checkPolindrom(string n){
	int len = n.length();
	for(int i=0; i<len/2; i++){
		if(n[i] != n[len - i - 1]){
			return 0;
		}
	}
	return true;
}

int countCharacterInString(char x, string n){
	int count=0;
	for(int i=0; i<n.length(); i++){
		if(n[i] == x) count++;
	}
	return count;
}

bool checkPseudoPolindrom(string n){
	char c = NULL;
	for(int i=0; i<n.length(); i++){
		if(c == n[i]) continue;
		int count = countCharacterInString(n[i], n);
		if(count%2 == 1){
			if(c) {
				return false;
			}
			c = n[i];
		}
	}
	return true;
}

bool checkQuasiPolindrom(string n){
	for(int i=0; i<n.length(); i++){
		int count = countCharacterInString(n[i], n);
		if(count>=2)return true;
	}
	return false;
}

int main(){
	string n;
	cin>>n;
	if(checkPolindrom(n)) cout<<"Polindrom";
	else if(checkPseudoPolindrom(n)) cout<<"PseudoPolindrom";
	else if(checkQuasiPolindrom(n)) cout<<"QuasiPolindrom";
	else cout<<"None";
}

// Verilmis setrin polindrom, pseudo polindrom yoxsa quasi polindrom oldugunu teyin eden program

File: Programlar/Task 1/123.py
def checkPolindrom(n):
    if n == n[::-1]: return True
    else: return False

def checkPseudoPolindrom(n):
    c = None
    for i in n:
        if c == i: continue
        if n.count(i)%2 == 1:
            if c: return False
            c = i
    else: return True

def checkQuasiPolindrom(n):
    for i in n:
        if n.count(i) >= 2: return True
    else:  return False

n = input()
if checkPolindrom(n): print("Polindrom")
elif checkPseudoPolindrom(n): print("PseudoPolindrom")
elif checkQuasiPolindrom(n): print("QuasiPolindorm")
else: print("None")

#Verilmiş sətrin polindrom, pseudo polindrom, yoxsa quasi polindrom olduğunu təyin edən program
